docs/standard/serialization/system-text-json/custom-contracts.md

```diff
@@ -0,0 +1,120 @@
+
+The following table shows the modifications you can make and how to achieve them.
+
+| Modification | Applicable `JsonTypeInfo.Kind` | How to achieve it | Example |
+| - | - | - | - |
+| Customize a property's value | `JsonTypeInfoKind.Object` | Modify the <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.Get?displayProperty=nameWithType> delegate (for serialization) or <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.Set?displayProperty=nameWithType> delegate (for deserialization) for the property. | [Increment a property's value](#example-increment-a-propertys-value) |
+| Add or remove properties | `JsonTypeInfoKind.Object` | [Serialize private fields](#example-serialize-private-fields) |
+| Conditionally serialize a property | `JsonTypeInfoKind.Object` | Modify the <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.ShouldSerialize?displayProperty=nameWithType> predicate for the property. | [Ignore properties with a specific type](#example-ignore-properties-with-a-specific-type) |
+| Customize number handling for a specific type | `JsonTypeInfoKind.None` | Modify the <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo.NumberHandling?displayProperty=nameWithType> value for the type. | [Allow int values to be strings](#example-allow-int-values-to-be-strings) |
+
+## Example: Increment a property's value
+
+Consider the following example where the modifier increments the value of a certain property on deserialization by modifying its <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.Set?displayProperty=nameWithType> delegate. Besides defining the modifier, the example also introduces a new attribute that it uses to locate the property whose value should be incremented. This is an example of *customizing a property*.
+
+:::code language="csharp" source="snippets/custom-contracts/SerializationCount.cs":::
+
+Notice in the output that the value of `RoundTrips` is incremented each time the `Product` instance is deserialized.
+
+## Example: Serialize private fields
+
+By default, `System.Text.Json` ignores private fields and properties. This example adds a new class-wide attribute, `JsonIncludePrivateFieldsAttribute`, to change that default. If the modifier finds the attribute on a type, it adds all the private fields on the type as new properties to <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo>.
+
+:::code language="csharp" source="snippets/custom-contracts/PrivateFields.cs":::
+
+> [!TIP]
+> If your private field names start with underscores, consider adding a JSON naming policy to remove the underscore and capitalize the name.
```

Comment:
I think users might misinterpret this as saying that `JsonSerializerOptions.PropertyNamingPolicy` is being applied to custom defined `JsonPropertyInfo` instances, whereas it is only being consulted by the built-in resolvers.

I would suggest changing the term "JSON naming policy" to something different to remove that connotation.